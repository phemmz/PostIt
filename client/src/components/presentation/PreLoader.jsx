import React from 'react';

/**
 * @description PreLoader Component
 * @return {*} div
 */
const PreLoader = () => {
  return (
    <div className="container">
      <div className="col s3 offset-s4 my-preloader">
        <div className="valign-wrapper preloader-wrapper active">
          <div className="spinner-layer spinner-red-only">
            <div className="circle-clipper left">
              <div className="circle" />
            </div>
            <div className="gap-patch">
              <div className="circle" />
            </div>
            <div className="circle-clipper right">
              <div className="circle" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default PreLoader;
