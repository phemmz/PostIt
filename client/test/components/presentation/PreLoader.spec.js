import React from "react";
import { mount } from "enzyme";
import PreLoader from '../../../src/components/presentation/PreLoader';

describe("PreLoader Component", () => {
  let mountedPreLoader;

  const preLoader = () => {
    if (!mountedPreLoader) {
      mountedPreLoader = mount(
        <PreLoader />
      );
    }
    return mountedPreLoader;
  }

  beforeEach(() => {
    preLoader: undefined;
  });

  it("should always render a div", () => {
    const divs = preLoader().find("div");
    expect(preLoader()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered in the div", () => {
    const divs = preLoader().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(preLoader().children());
  });
});
