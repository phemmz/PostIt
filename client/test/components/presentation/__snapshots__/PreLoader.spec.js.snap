// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreLoader Component should always render a div 1`] = `
<PreLoader>
  <div
    className="container"
  >
    <div
      className="col s3 offset-s4 my-preloader"
    >
      <div
        className="valign-wrapper preloader-wrapper active"
      >
        <div
          className="spinner-layer spinner-red-only"
        >
          <div
            className="circle-clipper left"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="gap-patch"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</PreLoader>
`;
